### PR TITLE
refactor: enable strict-boolean-expressions globally

### LINKED ISSUE #1046: Strict boolean expressions
> We probably need to scan the whole codebase for cases like these at some point – IIRC, there are a number of places where we check for "falsy" values instead of `undefined` or `null`. I think there is the possibility to make the eslint rules a bit stricter in this regard to help us with that.

_Originally posted by @JKRhb in https://github.com/eclipse-thingweb/node-wot/pull/1042#discussion_r1277735531_

As mentioned above, we are not currently verifying that our codebase uses the boolean expressions correctly. As we all know the nullish values can sometimes by tricked and led us to unwanted bugs (as the bugs that the PR above was solving). I tried to enable  `"@typescript-eslint/strict-boolean-expressions": "error"` into our eslint file, but that requires `strictNullChecks` to be enabled in `tsconfig.json`. Therefore, I suggest first completing the long standing #758 and then looking back to this issue. 

Packages to be fixed:
- [x] binding-coap
- [ ] binding-modbus
- [ ] browser-bundle
- [ ] binding-file
- [ ] binding-mqtt
- [ ] cli
- [ ] binding-netconf
- [x] core
- [ ] binding-opcua
- [x] examples
- [ ] binding-websocket
- [x] td-tools
- [x] binding-http
- [ ] binding-mbus

### PR DESCRIPTION
**Pull request by JKRhb**
Depends on #1126, resolves #1046.

### CODE REVIEW

**Review by danielpeintner — APPROVED** [review]
That's a great clean-up! 🎉

File: packages/cli/src/cli.ts [hunk 1]
```diff
@@ -307,17 +309,16 @@ const runScripts = async function (servient: DefaultServient, scripts: Array<str
     };
 
     // eslint-disable-next-line @typescript-eslint/no-var-requires
-    const inspector = require("inspector");
     if (debug && debug.shouldBreak) {
         // Activate inspector only if is not already opened and wait for the debugger to attach
-        !inspector.url() && inspector.open(debug.port, debug.host, true);
+        inspector.url() != null && inspector.open(debug.port, debug.host, true);
```
Comment (relu91) [inline packages/cli/src/cli.ts:314]: Should this be:
```suggestion
        inspector.url() == null && inspector.open(debug.port, debug.host, true);
```

Comment (JKRhb) [reply packages/cli/src/cli.ts:314]: Thank you, good catch!

**Comment by JKRhb** [comment]
Applied a last tiny improvement in [10f9498](https://github.com/eclipse-thingweb/node-wot/pull/1127/commits/10f949822632861ab238d7d181e04592f0880bcc) – hopefully, this one is now good to go :)

File: packages/cli/src/cli-default-servient.ts [hunk 2]
```diff
@@ -98,10 +98,7 @@ export default class DefaultServient extends Servient {
 
         // apply flags
         if (clientOnly) {
-            if (!this.config.servient) {
-                this.config.servient = {};
-            }
-            this.config.servient.clientOnly = true;
+            this.config.servient ??= { clientOnly };
```
Comment (relu91) [inline packages/cli/src/cli-default-servient.ts:101]: are you sure this works as the previous code? In my understanding previously if `this.config.servient` was already defined we just set `clientOnly` property to `true`. With the new code if it is defined we do not override it.

File: packages/cli/src/cli.ts [hunk 3]
```diff
@@ -339,7 +340,7 @@ const runScripts = async function (servient: DefaultServient, scripts: Array<str
         });
     } else {
         // Activate inspector only if is not already opened and don't wait
-        debug && !inspector.url() && inspector.open(debug.port, debug.host, false);
+        debug != null && inspector.url() != null && inspector.open(debug.port, debug.host, false);
```
Comment (relu91) [inline packages/cli/src/cli.ts:343]: spotted another:
```suggestion
        debug != null && inspector.url() == null && inspector.open(debug.port, debug.host, false);
```

File: packages/cli/src/cli-default-servient.ts (hunk 2 above)
Comment (JKRhb) [reply packages/cli/src/cli-default-servient.ts:101]: Oh, you are right, then we guess we need to keep something like this here:

```suggestion
            this.config.servient ??= {};
            this.config.servient.clientOnly = true;
            
```

**Review by relu91 — APPROVED** [review]
good to go 🥳

**Pull request merged**